File: backend/app/routers/reports.py
import logging

from fastapi import APIRouter, Depends, Query
from sqlmodel import Session

from app.db import get_session
from app.models.report import (
    AccountAttribute,
    BalanceSummaryHistoryResponse,
    BalanceSummaryItem,
)
from app.services.reports import get_balance_summary, get_balance_summary_history

logger = logging.getLogger(__name__)

router = APIRouter(tags=["reports"])


@router.get("/reports/balance-summary", response_model=list[BalanceSummaryItem])
def balance_summary(
    attribute: AccountAttribute,
    month: str,
    session: Session = Depends(get_session),
) -> list[BalanceSummaryItem]:
    return get_balance_summary(attribute, month, session)


@router.get(
    "/reports/balance-summary/history",
    response_model=BalanceSummaryHistoryResponse,
)
def balance_summary_history(
    attribute: AccountAttribute,
    from_: str = Query(alias="from"),
    to: str | None = None,
    session: Session = Depends(get_session),
) -> BalanceSummaryHistoryResponse:
    return get_balance_summary_history(attribute, from_, to, session)

File: frontend/src/api/reports.ts
import { client } from "./client";

export interface BalanceSummaryItem {
  group_key: string | null;
  balance_sum_usd: number;
}

export interface BalanceSummaryHistoryItem {
  month: string;
  group_key: string | null;
  balance_sum_usd: number;
}

export const getBalanceSummaryBySide = (month: string) =>
  client.get<BalanceSummaryItem[]>(
    `/reports/balance-summary?attribute=side&month=${month}`,
  );

export const getBalanceSummaryByTags = (month: string) =>
  client.get<BalanceSummaryItem[]>(
    `/reports/balance-summary?attribute=tags&month=${month}`,
  );

export const getBalanceSummaryHistory = (from: string, to: string) =>
  client.get<BalanceSummaryHistoryItem[]>(
    `/reports/balance-summary/history?attribute=side&from=${from}&to=${to}`,
  );
